PowerEditor/src/WinControls/AnsiCharPanel/asciiListView.cpp

```diff
@@ -390,30 +510,35 @@ void AsciiListView::setValues(int codepage)
 	for (int i = 0 ; i < 256 ; ++i)
 	{
 		constexpr size_t bufSize = 8;
-		TCHAR dec[bufSize];
-		TCHAR hex[bufSize];
-		TCHAR htmlNumber[bufSize];
+		constexpr size_t bufSizeHex = 9;
+		wchar_t dec[bufSize]{};
+		wchar_t hex[bufSize]{};
+		wchar_t htmlNumber[bufSize]{};
+		wchar_t htmlHexNumber[bufSizeHex]{};
 		generic_string htmlName;
 		swprintf(dec, bufSize, TEXT("%d"), i);
 		swprintf(hex, bufSize, TEXT("%02X"), i);
 		generic_string s = getAscii(static_cast<unsigned char>(i));
 
 		if (codepage == 0 || codepage == 1252)
 		{
-			if ((i >= 32 && i <= 126) || (i >= 160 && i <= 255))
+			if ((i >= 32 && i <= 126 && i != 45) || (i >= 160 && i <= 255))
```

Comment:
Why the value 45 is treated other way?

Reply:
because minus sign HTML name is `&minus;` which has unicode codepoints [U+2212](https://codepoints.net/U+2212)

Reply:
@ozone10 
Are you telling me that HTML `&minus` == `U+2212`?
But then what about `&#x2d;` ? It should represent `-`, shouldn't it?

Reply:
@donho 
yes, `&minus;` is `U+2212`.
`-` is hyphen-minus, which is on every keyboard, and is character which usually does not need to escape in HTML

On the other hand `−` minus sign is not usually on keyboard layout, but is used a lot.